### PR TITLE
Incremental Classifier

### PR DESCRIPTION
**Pull request by OneAndZero24**
- Incremental Classifier used instead of LocalHead
- Add scaling
- Drop LocalHead

### CODE REVIEW

File: src/model/inc_classifier.py [hunk 1]
```diff
@@ -64,15 +66,18 @@ def increment(self, new_classes: list[int]):
                 self.active_units[new_classes] = 1
 
         if old_nclasses != new_nclasses:
+            self.mul *= old_nclasses/new_nclasses
```
Comment (pkrukowski1) [inline src/model/inc_classifier.py:69]: This is scaled incorrectly. Please remove it and refer to the comment on line 79.

File: src/model/inc_classifier.py [hunk 2]
```diff
@@ -64,15 +66,18 @@ def increment(self, new_classes: list[int]):
...
+            for name, param in self.classifier.parameters():
                 param.data[:old_nclasses] = state_dict[name]
         self.out_features = new_nclasses
         
 
     def forward(self, x):
-        out = self.classifier(x)
+        x = self.classifier[0](x)
+        x *= self.mul
+        out = self.classifier[1](x)
```
Comment (pkrukowski1) [inline src/model/inc_classifier.py:80]: There should be x = x*old_nclasses/new_nclasses + x_min * (1 - old_nclasses/new_nclasses) in line 79. Without the shift, it simply does not work because it only shrinks the domain within which the hills are defined. I wrote a proof demonstrating that the proposed scaling is valid. Please check our Overleaf and reach out to me if you have any questions.

Comment (pkrukowski1) [inline src/model/inc_classifier.py:80]: We were considering 'sum(dim=2)' within the local classification head, but it should actually be 'sum(dim=1).' This was my mistake, and I apologize for the confusion.

**Review by pkrukowski1 — CHANGES_REQUESTED** [review]
I have a general comment. Please review the current version of the local layer. There was a minor issue—it was defined as a mapping from R^N → R^N. In general, the local layer can be applied as a transformation from R^N → R^M, which is a more general version. Please ensure this generality is preserved and do not remove it, please. I proposed this change in my PR: https://github.com/OneAndZero24/local-cl/pull/1.

**Review by pkrukowski1 — APPROVED** [review]
